uint8 temp0
uint8 temp1
uint8 temp2
uint8 temp3
uint8 temp4
uint8 temp5
uint8 temp6
uint8 temp7